.github/workflows/build-wasm_of_ocaml.yml

```diff
@@ -98,51 +121,81 @@ jobs:
         run: opam exec -- dune build @runtest-wasm --profile with-effects
 
       - name: Run Base tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
+        if: matrix.all_jane_street_tests
         working-directory: ./janestreet/lib/base
         run: opam exec -- dune runtest
 
       - name: Run Base bigstring tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
+        if: matrix.all_jane_street_tests
         working-directory: ./janestreet/lib/base_bigstring
         run: opam exec -- dune runtest
 
       - name: Run Core tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
+        if: matrix.all_jane_street_tests
         working-directory: ./janestreet/lib/core
         run: opam exec -- dune runtest
 
       - name: Run Bignum tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
+        if: matrix.all_jane_street_tests
         working-directory: ./janestreet/lib/bignum
         run: opam exec -- dune runtest
 
       - name: Run Bin_prot tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
+        if: matrix.all_jane_street_tests
         working-directory: ./janestreet/lib/bin_prot
         run: opam exec -- dune runtest
 
       - name: Run String_dict tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
+        if: matrix.all_jane_street_tests
         working-directory: ./janestreet/lib/string_dict
         run: opam exec -- dune runtest
 
       - name: Run Zarith tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
+        if: matrix.all_jane_street_tests
         working-directory: ./janestreet/lib/zarith_stubs_js
         run: opam exec -- dune runtest
 
       - name: Run Virtual_dom tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
+        if: matrix.all_jane_street_tests
         working-directory: ./janestreet/lib/virtual_dom
         run: opam exec -- dune runtest
 
       - name: Run Bonsai tests
-        if: ${{ matrix.ocaml-compiler < '5.' && matrix.separate_compilation }}
-        working-directory: ./janestreet/lib/bonsai
+        if: matrix.all_jane_street_tests
+        working-directory: ./janestreet/lib/bonsai_test
+        run: opam exec -- dune runtest
+
+      - name: Write node wrapper
```

Comment:
Can we do something inside `ci_setup.ml` instead ? we could update janestreet/dune-workspace  by adding a `binaries` section and dumping the node wrapper somewhere in the janestreet directory